test/test_dataloader.py

```diff
@@ -1647,5 +1650,32 @@ def test_ind_worker_queue(self):
                 self._run_ind_worker_queue_test(batch_size=batch_size, num_workers=num_workers)
 
 
+class TestChunkDataset(TestCase):
+    def setUp(self):
+        super(TestChunkDataset, self).setUp()
+        self.chunk_count=3
+        self.batch_size=5
+        self.cache_size=100
+        self.preloaders=1
+        self.chunk_sampler = torch._C.data.SequentialSampler(size=self.chunk_count)
```

Comment:
You only need these members to create the wrappers, so can we simply create them in the constructor of the wrappers.

Reply:
Which one?

`chunk_count` is used by `chunk_sampler`
`batch_size` is used by `example_sampler` and `ChunkDatasetOptions`
`cache_size` and `preloaders` are used by ChunkDatasetOptions

`SamplerWrapper` only gets the samplers instances (for chunk_sampler and example_sampler), no other input is used by the wrapper

Both chunk and example samplers can be created with different parameters depending on which Sampler specialization they use (sequential, random, distributed sequential, distributed random, etc), so it is not possible to embed all parameters in the wrapper class